lamindb/_record.py

```diff
@@ -740,14 +747,18 @@ def save(self, *args, **kwargs) -> Record:
                 revises._revises = None  # ensure we don't start a recursion
                 revises.save()
                 check_name_change(self)
+                check_key_change(self)
                 super(Record, self).save(*args, **kwargs)
                 _store_record_old_name(self)
+                _store_record_old_key(self)
```

Comment:
Rather than making a dedicated function `_store_record_old_key`, can you please just add the two lines here:
```
    if isinstance(record, (Artifact, Transform)):
        record._key = record.key
```

Easier to read and more performant. In parts, we have two many auxiliary functions.

Reply:
But _store_record_old_key is used in 3 places, that's why I made a function.

Reply:
Ok, fine!